File: clients/create/templates/ios16/Packages/Features/Login/Sources/Login/LoginRepository.swift
import Foundation

public enum LoginResult: Sendable, Equatable {
    case success(token: String)
    case failure(message: String)
}

public protocol LoginRepository: Sendable {
    func login(username: String, password: String) async -> LoginResult
}

/// 模板占位实现：接真实认证后端时替换本实现（协议与状态形状不动）。
public struct MockLoginRepository: LoginRepository {
    public init() {}

    public func login(username: String, password: String) async -> LoginResult {
        try? await Task.sleep(for: .milliseconds(600))
        if username.isEmpty || password.isEmpty {
            return .failure(message: "用户名或密码不能为空")
        }
        return .success(token: "mock-token")
    }
}

File: clients/create/templates/android-minsdk24/feature/login/src/main/kotlin/io/github/ydonghao/{{appPackageSegment}}/feature/login/LoginRepository.kt
package io.github.ydonghao.{{appPackageSegment}}.feature.login

import dagger.Binds
import dagger.Module
import dagger.hilt.InstallIn
import dagger.hilt.components.SingletonComponent
import javax.inject.Inject
import javax.inject.Singleton
import kotlinx.coroutines.delay

sealed interface LoginResult {
    data class Success(val token: String) : LoginResult
    data class Failure(val message: String) : LoginResult
}

interface LoginRepository {
    suspend fun login(username: String, password: String): LoginResult
}

/** 模板占位实现：接真实认证后端时替换（接口与 UiState 形状不动）。 */
@Singleton
class MockLoginRepository @Inject constructor() : LoginRepository {
    override suspend fun login(username: String, password: String): LoginResult {
        delay(600)
        return if (username.isBlank() || password.isBlank()) {
            LoginResult.Failure("用户名或密码不能为空")
        } else {
            LoginResult.Success("mock-token")
        }
    }
}

@Module
@InstallIn(SingletonComponent::class)
internal abstract class LoginModule {

    @Binds
    abstract fun bindLoginRepository(impl: MockLoginRepository): LoginRepository
}
